### ISSUE TITLE
CompserTask documentation

### ISSUE THREAD

**Issue by jawira** [phingofficial/phing#1163]
Hello, I have the following task:

```xml
  <target name="composer:validate" description="Check composer.json syntax">
    <composer>
      <arg value="validate"/>
      <arg value="--strict"/>
      <arg value="--no-check-lock"/>
    </composer>
  </target>
```

This task ran perfectly in `Phing 3.0.0-alpha2`, but in `Phing 3.0.0-alpha3` I get the following error message:

> Argument 1 passed to CommandlineArgument::setValue() must be of the type string, null given, called in phar:///usr/bin/phing/classes/phing/tasks/ext/ComposerTask.php on line 170

To solve this issue I have to refactor the target as this:

```xml
  <target name="composer:validate" description="Check composer.json syntax">
    <composer command="validate">
      <arg value="--strict"/>
      <arg value="--no-check-lock"/>
    </composer>
  </target>
```

I agree with `command` property being required, but if so then `ComposerTask`'s documentation and schema should be updated too. Also error message should be gracefully handled instead of printing the exception.

Let me know if you want me to do it :)

**Comment by siad007** [comment]
> I agree with command property being required, but if so then ComposerTask's documentation and schema should be updated too. Also error message should be gracefully handled instead of printing the exception.

You are totally right! We should mark the `command` attribute as required. Looking forward to see your pr soon. 😄

**Comment by jawira** [comment]
Coo, I will work on this next week :)

**Comment by jawira** [comment]
I realised  the `Exception` is not thrown by `ComposerTask`, but instead is thrown because *type hint* was added to `\CommandlineArgument::setValue`. Maybe this change will cause more exceptions in other tasks in the future.

Anyway, a PR fixing `ComposerTask` is available.

**Comment by mrook** [comment]
Fixed in #1175.
